stdlib/public/core/StringUTF8View.swift

```diff
@@ -317,6 +317,40 @@ extension String.UTF8View {
   }
 }
 
+extension String.UTF8View {
+
+  @available(SwiftStdlib 6.2, *)
+  public var span: Span<UTF8.CodeUnit> {
+    @lifetime(borrow self)
+    borrowing get {
+#if _runtime(_ObjC)
+      // handle non-UTF8 Objective-C bridging cases here
+      if !_guts.isFastUTF8 && _guts._object.hasObjCBridgeableObject {
```

Comment:
It occurred to me today that we may be able to do better here. `isFastUTF8` is not actually a binary property, even though most uses of it in practice can treat it that way. It's actually a 3-state thing: yes, no, yes-but-not-terminated.

If we're careful we should be able to handle that last case without using associated objects, which will make "NoCopy" bridged ASCII NSStrings fall into the happy path here.

Reply:
Here's the relevant bit of StringBridge.swift, we'd want a path that passes 0 instead of 1 as the argument

```
@_effects(readonly)
private func _NSStringASCIIPointer(_ str: _StringSelectorHolder) -> UnsafePointer<UInt8>? {
  //TODO(String bridging): Unconditionally asking for nul-terminated contents is
  // overly conservative and hurts perf with some NSStrings
  return unsafe str._fastCStringContents(1)?._asUInt8
}
```